### PR TITLE
feat(drawer-and-comps): added drawer and supporting components

### PR DESCRIPTION
**Pull request by mattnolting**
This PR fills in the gaps that exist between product implementation and PF Core/React. 
What it does:
- applies `pf-c-drawer` to layout with mobile support 
- added `button` to control `pf-c-drawer__panel` on mobile viewports [4687](https://github.com/patternfly/patternfly-react/issues/4687)
- adds concept of `sections` to form [3395](https://github.com/patternfly/patternfly/issues/3395)
- adds concept of `section-title` to form section [3395](https://github.com/patternfly/patternfly/issues/3395) 
- adds sub-tabs to tabbed section 
- adjusts `pf-c-title` `margin-bottom` for step titles
- adds concept of `pf-c-text/pf-c-title` modifiers [3426](https://github.com/patternfly/patternfly/issues/3246)
- changes `pf-c-expandable` to `pf-c-accordion` (you'll have to wire this up)
- changes custom tabs to default, secondary tabs

I'll create Core issues for most of the items listed above to provide better support for form layouts.

### CODE REVIEW

**Comment by AllenBW** [comment]
@mattnolting lil bitta linting to do before digging in! `npm run lint` should do the trick

**Comment by mattnolting** [comment]
@AllenBW doh!

File: src/Components/AnsibleTabs/AnsibleTabs.js [hunk 1]
```diff
@@ -121,123 +138,122 @@ InstallAnsibleTab.propTypes = {
...
+        <Tab eventKey={0} onClick={() => setActiveTabKey(0)} title={<TabTitleText>{intl.formatMessage(messages.roleVariables)}</TabTitleText>}>
+            <Grid hasGutter>
+                <TextContent>
+                    <Text component={TextVariants.h4}>
+                        {intl.formatMessage(messages.rolesAndConfig)}
+                    </Text>
+                    <Text> {intl.formatMessage(messages.rolesAndConfigDescription)} </Text>
+                </TextContent>
+                <Accordion asDefinitionList>
+                    {getSections(intl).map(section => <AccordionItem key={section.name} className='ins-c-install-client-tab__sections'>
```
Comment (LI0131) [inline src/Components/AnsibleTabs/AnsibleTabs.js:153]: I like the naming here but let's move it to the `Accordion` tag and update the sass

File: src/Components/AnsibleTabs/AnsibleTabs.scss [hunk 2]
```diff
@@ -5,91 +5,111 @@ section {
...
+
+.ins-c-install-client-tab__section {
+    padding-right: var(--pf-global--gutter);
+
+    .pf-c-expandable-section__toggle-text {
+        position: relative;
+        right: calc(var(--pf-global--spacer--sm) + var(--pf-global--spacer--xs));
+    }
+
+    .pf-c-expandable-section {
```
Comment (LI0131) [inline src/Components/AnsibleTabs/AnsibleTabs.scss:31]: all this `ExpandableSection` specific sass isn't really necessary with `Accordion`

File: src/Components/AnsibleTabs/AnsibleTabs.scss [hunk 3]
```diff
@@ -5,91 +5,111 @@ section {
...
             }
         }
-        .pf-c-expandable-section__content {
-            margin: var(--pf-global--spacer--md) 0 var(--pf-global--spacer--lg) 0;
-            p {
-                margin: var(--pf-global--spacer--md);
-            }
+    }
+
+    .pf-c-expandable-section__content {
```
Comment (LI0131) [inline src/Components/AnsibleTabs/AnsibleTabs.scss:60]: update to `pf-c-accordion__expanded-content-body`

File: src/Components/AnsibleTabs/AnsibleTabs.scss [hunk 4]
```diff
@@ -5,91 +5,111 @@ section {
...
-                width: 65%;
-                margin: 0 auto;
-            }
-        }
     }
-    .ins-c-install-ansible-tab__title {
-        margin-top: var(--pf-global--spacer--xl);
-        padding-left: var(--pf-global--gutter);
+
+    .pf-c-expandable-section__toggle-icon {
```
Comment (LI0131) [inline src/Components/AnsibleTabs/AnsibleTabs.scss:73]: We can take this out

File: src/Components/AnsibleTabs/AnsibleTabs.scss [hunk 5]
```diff
@@ -5,91 +5,111 @@ section {
...
+
+        p + p {
+            margin-top: var(--pf-global--spacer--md);
         }
     }
-}
 
-.ins-c-install-ansible-tab {
-    .ins-c-install-ansible-tab__downloads {
+    .pf-c-expandable-section__toggle {
```
Comment (LI0131) [inline src/Components/AnsibleTabs/AnsibleTabs.scss:68]: We can take this out

File: src/Components/AnsibleTabs/AnsibleTabs.scss [hunk 6]
```diff
@@ -5,91 +5,111 @@ section {
...
+    a {
+        text-decoration: none;
     }
-    .ins-c-install-client-tab__section {
-        padding-right: var(--pf-global--gutter);
-        hr {
-            margin-bottom: 0 !important;
+}
+
+.ins-c-install-client-tab__section {
```
Comment (LI0131) [inline src/Components/AnsibleTabs/AnsibleTabs.scss:23]: Update to `.ins-c-install-client-tab__sections`

File: src/Components/AnsibleTabs/AnsibleTabs.js [hunk 7]
```diff
@@ -3,10 +3,23 @@ import './AnsibleTabs.scss';
...
+    Divider,
+    Gallery,
+    Grid,
+    TextContent,
+    Tab,
+    Tabs,
+    TabTitleText,
+    Title,
+    TitleSizes
+} from '@patternfly/react-core';
```
Comment (AllenBW) [inline src/Components/AnsibleTabs/AnsibleTabs.js:22]: Can we get these imported from '/dist/ems/' makes for a smaller bundle size

File: src/Components/LayoutComponents/DrawerGroup.js [hunk 8]
```diff
@@ -0,0 +1,16 @@
+import React from 'react';
+import propTypes from 'prop-types';
```
Comment (AllenBW) [inline src/Components/LayoutComponents/DrawerGroup.js:2]: So this file, formstepgroup.js pagetitle.js and titlegroup.js are all the same level NDA file, applying Ah few classes to ah div, any chance we can make a single helper components that is passed the appropriate classes to accomplish this with far less code?

File: src/Routes/Register/Register.js [hunk 9]
```diff
@@ -1,75 +1,151 @@
...
+    DrawerCloseButton,
+    DrawerContent,
+    DrawerContentBody,
+    DrawerHead,
+    DrawerPanelBody,
+    DrawerPanelContent,
+    PageSection,
+    PageSectionVariants,
+    TextContent
+} from '@patternfly/react-core';
```
Comment (AllenBW) [inline src/Routes/Register/Register.js:31]: Again, import from '/dist/ems'

File: src/Routes/Register/Register.js [hunk 10]
```diff
@@ -1,75 +1,151 @@
...
+
+import {
+    ColumnsIcon
+} from '@patternfly/react-icons';
+
```
Comment (AllenBW) [inline src/Routes/Register/Register.js:36]: Also import the dist version

File: src/Routes/Register/Register.scss [hunk 11]
```diff
@@ -1,27 +1,42 @@
...
+        --pf-c-drawer__panel--FlexBasis: 40%;
+    }
+
+    @media screen and (min-width: $pf-global--breakpoint--xl) {
+        --pf-c-drawer__panel--FlexBasis: 35%;
+    }
+
+    .pf-c-drawer__panel {
+        pre {
+            background: #e5e5e5;
```
Comment (AllenBW) [inline src/Routes/Register/Register.scss:34]: Can ah pf variable be used here?

File: src/Components/LayoutComponents/DrawerGroup.js (hunk 8 above)
Comment (mattnolting) [reply src/Components/LayoutComponents/DrawerGroup.js:2]: Oh yeah, that makes sense!

File: src/Components/AnsibleTabs/AnsibleTabs.js (hunk 7 above)
Comment (mattnolting) [reply src/Components/AnsibleTabs/AnsibleTabs.js:22]: Yessir

**Comment by mattnolting** [comment]
Let's see how these updates work! 

@LI0131 I made your requested CSS updates and reorganized the files a bit. Also added titles to enhance readability/scanability. 

@AllenBW I combined the `title` and `group` components into one. Let me know it that's more of what you want.

Cheers!

**Pull request merged**